carbon-design-system/carbon
# fix(NumberInput): Allow empty input for NumberInput with allowEmpty prop (#15985)

Closes #15985
Closes #16016

There were multiple issues open for functionality in the NumberInput.

1. The input did not allow for an empty input, even with `allowEmpty = true`. `Number()` automatically assigns `0` to an empty string. A ternary was added to capture this instance and allow for the empty string when `allowEmpty = true`. It will still default to 0 when `allowEmpty = false`.
2. The input did not allow for negative input. I tested this in the [live demo](https://v10.carbondesignsystem.com/components/number-input/code/#live-demo) and Storybook. It works in both instances. This issue was likely due to the `min` value equaling 0. When `min` is updated to a negative value, the input allows for negatives.
    
    <img width="300" alt="Screenshot 2024-04-01 at 00 16 05" src="https://github.com/carbon-design-system/carbon/assets/4097509/5d23f59a-2c21-43b1-98f2-d40af1d8e1d2">

#### Changelog

**Changed**

- allow empty inputs when `allowEmpty = true`

#### Testing / Reviewing

1. Start Storybook
2. visit [NumberInput Playground on localhost:3000](http://localhost:3000/?path=/story/components-numberinput--playground&args=allowEmpty:!true;min:-200;).
3. Ensure the following args are updated:
    | prop | value | purpose |
    | ---- | ------ | -------- |
    | `allowEmpty` | `true` | Ensures that an empty value can be used |
    | `min` | `-200` | Allows for negative values. The default of 0 causes an error to display. |
4. With these params set, select then delete the input value. It should be an empty input.
5. Then enter a negative value greater than the `min` value. Should allow the value and not show an error.
